### PR TITLE
Enhance dataset with Spanish nicknames

### PR DESCRIPTION
**Pull request by Ahosseinzadeh723**
Changes include:

- Addition of Spanish nicknames for male and female names.
- Addition of English version for Spanish names

### CODE REVIEW

File: names.csv [hunk 1]
```diff
@@ -1049,7 +1049,7 @@ tasha,tash,tashie
 ted,teddy
 temperance,tempy
 terence,terry
-teresa,terry,tess,tessa,tessie
+teresa,terry,tess,tessa,tessie,tere,theresa
```
Comment (carltonnorthern) [inline names.csv:1052]: I could be wrong but I don't think "Theresa" would be a nickname or diminutive name for "Teresa".

File: names.csv [hunk 2]
```diff
@@ -1144,3 +1144,96 @@ zack,zach,zak
...
+alfredo,fredi,alfred
+andres,andrew
+anibal,hannibal
+antonio,tono,tony,anthony
+armando,nando,herman
+arturo, arthur
+bernardo,berna,bernard
+candelario,candel
+carlos,charles
+cristian,cris,christian
```
Comment (carltonnorthern) [inline names.csv:1159]: Christian and Cristan falls under the same case as Theresa and Teresa.

File: names.csv [hunk 3]
```diff
@@ -1144,3 +1144,96 @@ zack,zach,zak
...
+cristobal,cris,christopher
```
Comment (carltonnorthern) [inline names.csv:1160]: Is Christopher a nickname for Cristobal?

File: names.csv [hunk 4]
```diff
@@ -1144,3 +1144,96 @@ zack,zach,zak
...
+delfino,pillo
+domingo,mingo
+eduardo,lalo,edward
+enrique,quique,henry
+ernesto,neto,ernest
+esteban,steven
+federico,quiquo,kiko,federick
+fernando,nando,ferdinand
+francisco,paco,pancho,francis
```
Comment (carltonnorthern) [inline names.csv:1169]: I have a dog named "Paco". I didn't realize that was a nickname for Francisco. I'll have to start calling him that. :)

File: names.csv [hunk 5]
```diff
@@ -1144,3 +1144,96 @@ zack,zach,zak
...
+gelardo,gera,harold,gerald
+guadalupe,lupe,lupillo
+guillermo,memo,william
+hector,teto
+hipolito,polin,hippolytos
+humberto,beto,hubert
+ignacio,nacho,ignatitus
+jaime,james
+jesus,chuy,chucho
+jorge,koque,george
```
Comment (carltonnorthern) [inline names.csv:1179]: I've noticed a few of these where the Spanish name has an equivalent English name. I'm on the fence if we want to include these or not. I think it makes sense if a common convention would be for a Jorge to go by George in a more English setting.

File: names.csv [hunk 6]
```diff
@@ -1144,3 +1144,96 @@ zack,zach,zak
 zebedee,zeb
 zedediah,dyer,zed,diah
 zephaniah,zeph
+adam,adan
```
Comment (carltonnorthern) [inline names.csv:1147]: We'll need these additions included in the list alphabetically.

**Comment by carltonnorthern** [comment]
This is a great addition @Ahosseinzadeh723 !  I've left a comments to resolve before we can merge it in.

File: names.csv (hunk 1 above)
Comment (carltonnorthern) [reply names.csv:1052]: Oh but Theresa would be the English version of Teresa?

**Comment by NickCrews** [comment]
@Ahosseinzadeh723 Thanks for the contribution! I can see why this would be useful.

@carltonnorthern I am hesitant to merge this in without thinking about it a bit more. Semantically, this is changing the meaning of this package lot. Before, it was just nicknames within english. Now we are blurring the line with spanish translations (even if these names are very common in english-speaking countries). I can see the immense usefulness of including this data, but I also worry that this will break other existing users, who might not want these translations included.

What if we took this as an opportunity to restructure the representation:

Currently we use a "wide" format of "canoncial,nickname1, nickname2,...,nicknameN". What if we switched to a normalized "long" format of "canonical,nickname"? eg "charles,charlie,carlos" would be

```
charles,charlie
charles,carlos
```

This would have the benefit of
1. cleaner diffs when editing. Entire lines would get added, deleted
2. Make it easier for consuming libraries to ingest the data, eg we could add a `pairs(self) -> Iterable[NamedTuple]` method to NickNamer. Then we could pass the data right into a pandas dataframe, or a normalized SQL table. I already do this in a [downstream library](https://github.com/NickCrews/mismo/blob/06c8f91a2b83a93e9728fc159373cb8f1493b583/mismo/lib/name/_nicknames.py#L86C5-L114), and but I have to access the private attributes of the NickNamer to do it and its sorta gross.

This format change also opens the door for us to add per-edge attributes. For example, we could label these cross-language nicknames by adding a 3rd column:

```
canonical,nickname,languages
charles,charlie,EN:EN
charles,carlos,EN:SP
```

and then we could support filtering by these languages. And we could add more attributes in the future, for example

This would be breaking for the raw consumers of the CSV, but we could keep the python API stable. I think this would be worth it for the long-term life of this project. What do you think @carltonnorthern @Ahosseinzadeh723 ?

File: names.csv (hunk 3 above)
Comment (Ahosseinzadeh723) [reply names.csv:1160]: "Cristóbal" is the Spanish equivalent of "Christopher." Spanish-speaking people who move to the U.S. often adopt a more English-friendly name, and this is one of those cases.

**Comment by carltonnorthern** [comment]
@NickCrews I think this is an excellent idea. I had similar concerns but couldn't think of a way to handle these issues.

FYI, sorry for the slow response. I'm in the middle of orchestrating a move cross country.

**Comment by NickCrews** [comment]
I accidentally nuked the git history while getting auto-fix working in CI. It's back now, but that seemed to have closed this PR. Once/if I make this structural change, then we should re-open this PR with that new format

**Pull request closed without merging**